hydrogym/core.py

```diff
@@ -291,111 +289,111 @@ def solve(
+        self.solver: TransientSolver = env_config.get("solver")(self.flow, **env_config.get("solver_config", {}))
+        self.callbacks: Iterable[CallbackBase] = env_config.get("callbacks", [])
+        self.max_steps: int = env_config.get("max_steps", int(1e6))
+        self.iter: int = 0
+        self.q0: self.flow.StateType = self.flow.copy_state()
+
+        self.observation_space = gym.spaces.Box(
+            low=-np.inf,
+            high=np.inf,
+            shape=(self.flow.num_outputs,),
+            dtype=float,
+        )
+        self.action_space = gym.spaces.Box(
+            low=-self.flow.MAX_CONTROL,
+            high=self.flow.MAX_CONTROL,
+            shape=(self.flow.num_inputs,),
+            dtype=float,
+        )
+
+    def set_callbacks(self, callbacks: Iterable[CallbackBase]):
+        self.callbacks = callbacks
+
+    def step(self, action: Iterable[ArrayLike] = None) -> Tuple[ArrayLike, float, bool, dict]:
+        """Advance the state of the environment.  See gym.Env documentation
 
         Args:
             action (Iterable[ArrayLike], optional): Control inputs. Defaults to None.
 
         Returns:
             Tuple[ArrayLike, float, bool, dict]: obs, reward, done, info
         """
-    self.solver.step(self.iter, control=action)
-    self.iter += 1
-    t = self.iter * self.solver.dt
-    for cb in self.callbacks:
-      cb(self.iter, t, self.flow)
-    obs = self.flow.get_observations()
+        self.solver.step(self.iter, control=action)
+        self.iter += 1
+        t = self.iter * self.solver.dt
+        for cb in self.callbacks:
+            cb(self.iter, t, self.flow)
+        obs = self.flow.get_observations()
+
+        reward = self.get_reward()
+        done = self.check_complete()
+        info = {}
+
+        # Truncation=False as we handle it through max_steps for now
+        truncated=False
 
-    reward = self.get_reward()
-    done = self.check_complete()
-    info = {}
+        obs = self.stack_observations(obs)
 
-    obs = self.stack_observations(obs)
+        return obs, reward, done, truncated, info
 
-    return obs, reward, done, info
+    # TODO: Use this to allow for arbitrary returns from collect_observations
+    #  That are then converted to a list/tuple/ndarray here
+    def stack_observations(self, obs):
+        return obs
 
-  # TODO: Use this to allow for arbitrary returns from collect_observations
-  #  That are then converted to a list/tuple/ndarray here
-  def stack_observations(self, obs):
-    return obs
+    def get_reward(self):
+        return -self.solver.dt * self.flow.evaluate_objective()
 
-  def get_reward(self):
-    return -self.solver.dt * self.flow.evaluate_objective()
+    def check_complete(self):
+        return self.iter > self.max_steps
 
-  def check_complete(self):
-    return self.iter > self.max_steps
+    def reset(self, seed=None, t=0.0, options=None) -> Union[ArrayLike, Tuple[ArrayLike, dict]]:
```

Comment:
@jcallaham if you already want to have a look at this one function in the core abstraction - I had to extend it to be compliant with the modern gymnasium spec. Specifically the passing in of the seed, and the options, and the return of the empty info dict